src/yunohost/io.py

```diff
@@ -0,0 +1,222 @@
+                              m18n.n('io_cannot_write_file', file=file_path))
+    except Exception as e:
+        raise MoulinetteError(errno.EIO,
+                              m18n.n('io_unknown_error_writing_file',
+                                     file=file_path, error=str(e)))
+
+
+def _remove_file(file_path):
+    """
+    Remove a regular file if it exists
+
+    Keyword argument:
+        file_path -- Path of the file to remove
+    """
+    # Assumptions
+    assert isinstance(file_path, str)
+
+    if os.path.exists(file_path):
+        try:
+            os.remove(file_path)
+        except Exception as e:
+            raise MoulinetteError(errno.EIO,
+                                  m18n.n('io_unknown_error_removing_file',
+                                         file=file_path, error=str(e)))
+
+
+def _set_permissions(file_path, user, group, permissions):
+    """
+    Change permissions of a given file.
+    Example : _set_permissions("/etc/swag", "root", "www-data", 0750)
+
+    Keyword argument:
+        file_path -- Path to the json file
+        user -- Some user name
+        group -- Some unix group name
+        permissions -- Permissions to set, preferrably in octal format
+    """
+    # Assumptions
+    assert isinstance(file_path, str)
+    assert isinstance(user, str)
+    assert isinstance(group, str)
+    assert isinstance(permissions, int)
+
+    # Check file exist
+    if not os.path.isfile(file_path):
+        raise MoulinetteError(errno.EROENT,
+                              m18n.n('io_no_such_file', file=file_path))
+
+    uid = pwd.getpwnam(user).pw_uid
+    gid = grp.getgrnam(group).gr_gid
+
+    os.chown(file_path, uid, gid)
+    os.chmod(file_path, permissions)
+
+
+def _download_text_data(url):
+    """
+    Download text from a url and returns the raw text
+
+    Keyword argument:
+        url -- The url to download the data from
+    """
+    # Assumptions
+    assert isinstance(file_path, str)
+
+    # Download file
+    try:
+        r = requests.get(url, timeout=30)
+    # SSL exceptions
+    except requests.exceptions.SSLError:
+        raise MoulinetteError(errno.EBADE,
+                              m18n.n('io_download_ssl_error',
+                                     url=url))
+    # Timeout exceptions
+    except requests.exceptions.Timeout:
+        raise MoulinetteError(errno.ETIME,
+                              m18n.n('io_download_timeout',
+                                     url=url))
+    # Unknown stuff
+    except Exception:
+        raise MoulinetteError(errno.ECONNRESET,
+                              m18n.n('io_download_unknown_error',
+                                     url=url))
+    # Assume error if status code is not 200 (OK)
+    if r.status_code != 200:
+        raise MoulinetteError(errno.EBADE,
+                              m18n.n('io_download_bad_status_code',
+                                     url=url, code=r.status_code))
+
+    return r.text
```

Comment:
Maybe we could provide a wrapper around `_download_text_data()` that download, parse and return valid JSON ?